### PR TITLE
feat: add terraform and jenkins template

### PR DESCRIPTION
**Pull request by spartan-toannhan**
## Description

- Add the Terraform and Jenkins definitions for the React application
  - Jenkins: Use the Jenkins shared library by adding the Jenkinsfile
  - Terraform: use Terragrunt to define the static website

https://c0x12c.atlassian.net/browse/SC-25

### Type of change

- [ ] 🕷 Bug fix
- [x] 💎 New feature
- [ ] ⚙ Maintenance

### Changes

- _List changes here so reviewers know what is important to check..._

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have tested that the feature or bug fix works as expected
- [x] I have included helpful comments, particularly in hard-to-understand areas
- [ ] I have added tests that prove my changes are functioning
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: terraform/environments/prod/env.hcl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+locals {
+  environment                = "prod"
+  dns_name                   = "REPLACE_ME"
+  route53_zone_id            = "REPLACE_ME"
+  route53_zone_name          = "REPLACE_ME"
+  global_tls_certificate_arn = "REPLACE_ME"
```
Comment (spartan-dainguyen) [inline terraform/environments/prod/env.hcl:6]: Is this something we have to manually create and REPLACE here?

**Comment by spartan-dainguyen** [comment]
I think you will need to leave a README so the dev can see steps to deploy an react app.

File: terraform/environments/prod/env.hcl (hunk 1 above)
Comment (spartan-toannhan) [reply terraform/environments/prod/env.hcl]: that's right @spartan-dainguyen , this is specific to each project, we cannot define the value here

**Review by spartan-dainguyen — APPROVED** [review]
LGTM

**Pull request merged**